controllers/userController.js

```diff
@@ -434,17 +485,30 @@ function parseErrors(errors) {
 
  			mailer.send(mail.to, mail.subject, mail.text)
  			.then(function() {
- 				req.flash('info', {
- 					message: 'An e-mail has been sent to ' + user.email + ' with further instructions.'
- 				})
+				if (!wantJSON) {
+					req.flash('info', {
+						message: 'We emailed further instructions to ' + user.email + '.'
+					})
+				}
  				done()
  			})
  			.catch(done)
  		}
  		], function(err) {
- 			if (err)
+ 			if (err) {
+				res.status(500).json({ok:false, message: 'The server could not email you. Please contact us for assistance.'})
  				return next(err)
+			}
+			if (wantJSON) {
+				return res.status(200).json({
+					ok:		true,
```

Comment:
no particular reason. I prefer to use 'success', but found too many things using 'ok'. Since this is backend, and we haven't settled on the format of responses (#596), I didn't want to touch too much.